src/components/BlogPostListMenu.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <div class="menu-container">
+    <div
+      class="menu-item"
+      :class="{ 'active-menu-item': item === props.activeItem }"
+      v-for="item in props.menuItems"
+      @click="props.onPressMenuItem?.(item)"
+      :key="item"
+    >
+      {{ item }}
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { PropType } from 'vue'
+
+const props = defineProps({
+  onPressMenuItem: {
+    type: Function,
```

Comment:
How to pass a Function with typed arguments, ie: `type: (item: string) => void`?

Reply:
try to search the repo, but it's possible

Reply:
types are in websrc/src/entities
--when too many or reused in at least 2 different places